packages/core/src/routes/hook.ts

```diff
@@ -21,21 +24,73 @@ export default function hookRoutes<T extends AuthedRouter>(
     }
   );
 
+  router.get(
+    '/hooks/:id',
+    koaGuard({
+      params: z.object({ id: z.string().min(1) }),
+      response: Hooks.guard,
+      status: [200, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+
+      ctx.body = await findHookById(id);
+
+      return next();
+    }
+  );
+
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, name: true, signingKey: true, enabled: true, events: true })
+        .extend({
+          name: string().min(1).optional(),
+          events: hookEventsGuard.nonempty().optional(),
+        }),
+      response: Hooks.guard,
+      status: [200, 422],
+    }),
     async (ctx, next) => {
+      const { name, event, events, config, ...rest } = ctx.guard.body;
+
+      const hasConflictEventFields = event && events;
+      assertThat(
+        !hasConflictEventFields,
+        new RequestError({ code: 'hook.conflict_event_fields', status: 422 })
+      );
+
+      const eventsToCreate = events ?? conditional(event && [event]);
+      assertThat(eventsToCreate, new RequestError({ code: 'hook.missing_events', status: 422 }));
+
       ctx.body = await insertHook({
+        ...rest,
         id: generateStandardId(),
-        ...ctx.guard.body,
+        /**
+         * Note:
+         * In the next hook update, the `name` field will be not nullable.
+         * We need to ensure the new created hook name is not null or empty.
+         */
+        name: name ?? 'unnamed',
+        event: event ?? null,
+        events: eventsToCreate,
+        signingKey: generateStandardId(),
+        enabled: true,
+        config: {
+          ...config,
+          retries: config.retries ?? 3,
+        },
       });
 
       return next();
     }
   );
 
-  router.get(
-    '/hooks/:id',
+  router.post(
+    '/hooks/:id/signing-key',
     koaGuard({
       params: z.object({ id: z.string().min(1) }),
```

Comment:
nit: does /hooks//signing-key a valid path? do we need to guard the id  min length here?

Reply:
`/hooks//signing-key` is not a valid path, so we need to limit the id min(1).